### PR TITLE
cmake: fix checking version of libbcc

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/vltrace/40)

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -34,7 +34,16 @@ project(vltrace C)
...
-pkg_check_modules(LIBBCC REQUIRED libbcc)
+
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(LIBBCC REQUIRED libbcc)
+else()
+	find_package(LIBBCC QUIET REQUIRED)
+endif()
+
+if(NOT LIBBCC_FOUND)
+	message(FATAL_ERROR "libbcc not found - needed by vltrace")
```
Comment (marcinslusarz) [inline src/CMakeLists.txt]: This message will never be printed.

**Comment by krzycz** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40)*

File: src/CMakeLists.txt (hunk 1 above)
Comment (ldorau) [reply src/CMakeLists.txt]: @marcinslusarz Fixed

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -31,10 +31,40 @@
...
+
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(LIBBCC REQUIRED libbcc)
+else()
+	find_package(LIBBCC QUIET REQUIRED)
+endif()
+
+if(LIBBCC_FOUND)
+	set(ORIG_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
+	set(CMAKE_REQUIRED_LIBRARIES "${ORIG_CMAKE_REQUIRED_LIBRARIES} -lbcc")
```
Comment (GBuella) [inline src/CMakeLists.txt]: Well, if you save the original value, then probably you meant to add this after the check:
`set(CMAKE_REQUIRED_LIBRARIES "${ORIG_CMAKE_REQUIRED_LIBRARIES}")`

**Comment by krzycz** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40)*

**Comment by ldorau** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/CMakeLists.txt, line 48 at r2](https://reviewable.io:443/reviews/pmem/vltrace/40#-KwL8bCWaPjQrqvfHaXW-r2-48:-KwnLf90ZYh1UmMUURYw:b-bxnlmm) ([raw file](https://github.com/pmem/vltrace/blob/8f09f060fdf937120ace5f6e373942527aaa6e29/src/CMakeLists.txt#L48)):*
<details><summary><i>Previously, GBuella (Gabor Buella) wrote…</i></summary><blockquote>

Well, if you save the original value, then probably you meant to add this after the check:
`set(CMAKE_REQUIRED_LIBRARIES "${ORIG_CMAKE_REQUIRED_LIBRARIES}")`
</blockquote></details>

Well, in fact saving is not needed. I removed it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40)*

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -31,10 +31,38 @@
...
+
 add_cstyle(src)
 
 find_package(PkgConfig)
-pkg_check_modules(LIBBCC REQUIRED libbcc)
+
+if(PKG_CONFIG_FOUND)
+	pkg_check_modules(LIBBCC REQUIRED libbcc)
+else()
+	find_package(LIBBCC QUIET REQUIRED)
```
Comment (marcinslusarz) [inline src/CMakeLists.txt]: QUIET *and* REQUIRED? How is this supposed to work?

Comment (ldorau) [reply src/CMakeLists.txt]: It is strange, but it works :-) anyway I fixed it.

**Comment by marcinslusarz** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40#-:-KwoVpb9Gax5cubezD2v:bnfp4nl)*

**Comment by ldorau** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/CMakeLists.txt, line 48 at r2](https://reviewable.io:443/reviews/pmem/vltrace/40#-KwL8bCWaPjQrqvfHaXW-r2-48:-Kx7eRReQZfDsA4Z6hRJ:b-6yoyb6) ([raw file](https://github.com/pmem/vltrace/blob/8f09f060fdf937120ace5f6e373942527aaa6e29/src/CMakeLists.txt#L48)):*
<details><summary><i>Previously, ldorau (Lukasz Dorau) wrote…</i></summary><blockquote>

Well, in fact saving is not needed. I removed it.
</blockquote></details>

@GBuella, it is fixed, please resolve

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40)*

**Comment by ldorau** [comment]
@GBuella ping

**Comment by GBuella** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/vltrace/40#-:-KxN1DjqvgZAF1E_yyAo:bnfp4nl)*

**Comment by ldorau** [comment]
@marcinslusarz ping :-)

**Pull request merged**